Dockerfile.apiserver

```diff
@@ -3,21 +3,24 @@ ARG BASE_IMAGE
 FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.19-alpine as builder
 ARG GOPROXY
 ENV GOPROXY=${GOPROXY:-https://goproxy.cn}
-WORKDIR /workspace
+WORKDIR /cache
```

Comment:
这一修改基于 Docker build 的  cache 机制，可以节省构建时间。
若 mod 下载使用单独的文件夹，只要 mod 文件不修改，Docker build 的 cache 机制可以复用已经完成 mod 下载的 layer 层，无需再次进行依赖下载。
若使用同一文件夹，在 mod 文件未修改但代码修改的情况下，由于 workspace 下内容有变动，镜像构建时每次都要去执行 go mod download 去下载依赖，影响构建时间。

Reply:
> 在 mod 文件未修改但代码修改的情况下，由于 workspace 下内容有变动，镜像构建时每次都要去执行 go mod download 去下载依赖，影响构建时间。

I don't think so. If the mod file is not changed, it will not re-download go modules.

Reply:
> > 在 mod 文件未修改但代码修改的情况下，由于 workspace 下内容有变动，镜像构建时每次都要去执行 go mod download 去下载依赖，影响构建时间。
> 
> I don't think so. If the mod file is not changed, it will not re-download go modules.

我的构建服务器环境的问题，镜像分层构建时由于中间层 tag 为 <none>，我们内部的构建服务器把 tag 为 <none> 镜像会定时清理导致我误以为是 Dockerfile 的问题，多谢解答